### PR TITLE
Patch Block.isVisuallyOpaque() to be state-aware

### PR DESCRIPTION
**Pull request by bs2609**
This adds an override of `isVisuallyOpaque()` that takes an `IBlockState` parameter. Tried to keep changes as small as possible.

### CODE REVIEW

**Comment by williewillus** [comment]
I'd name it properly too, isVisuallyOpaque is a terrible name:

https://github.com/ModCoderPack/MCPBot-Issues/issues/289

**Comment by bs2609** [comment]
OK, renamed method to `causesSuffocation`, and added notes about usage from the [MCP issue](https://github.com/ModCoderPack/MCPBot-Issues/issues/289).

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -274,12 +274,16 @@
...
++     * <p>
++     * This method is used for:
++     * <br> Checking if an entity should be suffocating
++     * <br> Checking if the suffocation overlay should be rendered in 1st person
 +     *
 +     * @param state The current state
-+     * @return True if the block is visually opaque
++     * @return True if the block can suffocate entities that are inside it.
 +     */
-+    public boolean isVisuallyOpaque(IBlockState state)
```
Comment (bs2609) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: This is the newly-added version that takes an `IBlockState` parameter. The original still exists and has been marked deprecated, see patch line 33.

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 2]
```diff
@@ -266,6 +274,21 @@
...
++     * State-sensitive version of func_176214_u.
++     * <p>
++     * This method is used for:
++     * <br> Checking if an entity should be suffocating
++     * <br> Checking if the suffocation overlay should be rendered in 1st person
++     *
++     * @param state The current state
++     * @return True if the block can suffocate entities that are inside it.
++     */
++    public boolean causesSuffocation(IBlockState state)
```
Comment (mezz) [inline patches/minecraft/net/minecraft/block/Block.java.patch:286]: I'm pretty sure this new function is exactly the same as `Block.func_149637_q`.

**Review by mezz — CHANGES_REQUESTED** [review]
Looks good but I think your new method already exists.

File: patches/minecraft/net/minecraft/block/Block.java.patch (hunk 2 above)
Comment (bs2609) [reply patches/minecraft/net/minecraft/block/Block.java.patch:286]: That's certainly possible, I need to have a proper look at 1.11 code.
I'll close this once I've done that if Mojang sorted this themselves.

Comment (bs2609) [reply patches/minecraft/net/minecraft/block/Block.java.patch:286]: `func_149637_q` is `isBlockNormalCube`. The default logic used is the same, but they are called from different places and overriden differently.

Comment (bs2609) [reply patches/minecraft/net/minecraft/block/Block.java.patch:286]: OK, the actual 1.11 methods are `func_191058_s` (`IBlockProperties.isViewBlocking`) and `func_176214_u` (`Block.causesSuffocation`). Mojang didn't update the default implementation though >_>

**Comment by bs2609** [comment]
Related: https://github.com/ModCoderPack/MCPBot-Issues/issues/311

**Comment by mezz** [comment]
I'm very against adding more "is an opaque block with various micro-differences" methods to `Block.java`, it's already very difficult to navigate.
Is the existing method good enough for this purpose?

**Comment by bs2609** [comment]
What's in 1.11 is Good Enough(tm), the MCP naming is the bigger issue there.

I'll close this, it's a bit too late for 1.10.x now.

Thanks for the review.

**Comment by mezz** [comment]
Cool, thanks.

**Pull request closed without merging**